### PR TITLE
Disallow >1 group to suspend

### PR DESCRIPTION
**Pull request by hakonhall**
* If there is more than one group, disallow suspending a node if there is a node in another group that has a user wanted state != UP.
* If there is 1 group, disallow suspending more than 1 node.

In addition, we no longer allow suspension if a node is already in maintenance, as maintenance is used with grace period exactly to defer downing of node.

### CODE REVIEW

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeCheckerTest.java [hunk 1]
```diff
@@ -139,6 +146,158 @@ public void testUnknownStorageNode() {
...
+        assertThat(result.getReason(), is("At most one node can have a wanted state when #groups = 1: " +
+                "Other storage node 0 has wanted state Maintenance"));
+    }
+
+    @Test
+    public void testWhenOtherDistributorIsDown() {
+        // Nodes 0-3, storage node 0 being in maintenance with "Orchestrator" description.
+        ContentCluster cluster = createCluster(createNodes(4));
+        cluster.clusterInfo().getDistributorNodeInfo(0)
+                .setWantedState(new NodeState(NodeType.DISTRIBUTOR, State.DOWN).setDescription("Orchestrator"));
```
Comment (vekterli) [inline clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeCheckerTest.java:174]: Should we test without the explicit `Orchestrator` reason for at least one test? I seem to recall we have some special logic for that particular string.

**Review by vekterli — APPROVED** [review]
:+1:

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeCheckerTest.java (hunk 1 above)
Comment (hakonhall) [reply clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeCheckerTest.java:174]: "Orchestrator" isn't hard-coded in the cluster controller, but the rule "allow suspending if another node in the group is suspended" finds Maintenance nodes with the same description, hence would find "Orchestrator" Maintenance storage nodes.  This match on the description is tested in testDifferentDescriptionImpliesDenied.

The description is set to Orchestrator here mostly for documentation purposes - I could remove it or use a random description.

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeCheckerTest.java [hunk 2]
```diff
@@ -139,6 +146,158 @@ public void testUnknownStorageNode() {
         assertThat(result.getReason(), is("Unknown node storage.10"));
     }
 
+    @Test
+    public void testWhenOtherStorageNodeIsSuspended() {
```
[the remark this answers is not in the document]
Comment (hakonhall) [reply clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeCheckerTest.java]: Test names updated.

**Review by vekterli — APPROVED** [review]
:+1:

**Comment by hmusum** [comment]
Looks like system test https://github.com/vespa-engine/system-test/blob/master/tests/vds/teststatetools.rb started failing after this PR, might be that test just needs to be updated, haven't looked closely at it.

**Comment by hakonhall** [comment]
Message needs to be updated, fix in https://github.com/vespa-engine/system-test/pull/1496.

**Pull request merged**